indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -673,18 +698,30 @@ public Task getTask()
     {
       return task;
     }
+
+    @Override
+    public TaskLocation getLocation()
+    {
+      if (processHolder == null) {
+        return TaskLocation.unknown();
+      } else {
+        return TaskLocation.create(processHolder.host, processHolder.port);
+      }
+    }
   }
 
   private static class ProcessHolder
   {
     private final Process process;
     private final File logFile;
+    private final String host;
     private final int port;
 
-    private ProcessHolder(Process process, File logFile, int port)
+    private ProcessHolder(Process process, File logFile, String host, int port)
```

Comment:
going along with prior comment, can this take a HostAndPort argument instead of a host and port separate?